src/pages/survey/answer/[surveyId]/index.tsx

```diff
@@ -28,6 +29,12 @@ function SurveyResultsPage() {
     endDate,
   } = useSurveyResultsManager();
 
+  const [showOnlyWrittenResponses, setShowOnlyWrittenResponses] = useState(false);
+
+  const filteredAnswersData = showOnlyWrittenResponses
```

Comment:
It is not good approach. In the given solution, filtering will take place with each rerender of the component, which will be not efficient. It's better to create person state for `filteredAnswersData` and update it in `useEffect`.
After that all logic should be moved to `SurveyResultsManager`